src/confluent_kafka/schema_registry/avro.py

```diff
@@ -169,10 +170,8 @@ def __init__(self, schema_registry_client, schema_str,
             raise ValueError("Unrecognized properties: {}"
                              .format(", ".join(conf_copy.keys())))
 
-        # convert schema_str to Schema instance
-        schema = _schema_loads(schema_str)
         schema_dict = loads(schema.schema_str)
-        parsed_schema = parse_schema(schema_dict)
+        parsed_schema = parse_schema(schema_dict, _named_schemas=self._named_schemas)
```

Comment:
cool - fastavro seems like it has good support for referenced schemas, we should definitely support it.

again, the first thing i'm wondering is what type the API should expect. feel free to give some commentary on that, if you have a clear opinion.

Reply:
Using a Schema object seems reasonable and is done in the Java API:  https://docs.confluent.io/platform/current/schema-registry/serdes-develop/serdes-avro.html#avro-serializer.

However it does break backwards compatibility and does require awkwardly indicating 'AVRO' Schema type.   Also there are confusingly lots of lower level APIs (AVRO and fastavro) leaking through the Kafka API resulting in situations like many different types of Schemas:

   - [AVRO Schema](https://avro.apache.org/docs/current/gettingstartedpython.html)
   - [fastavro Schema](https://fastavro.readthedocs.io/en/latest/schema.html) - a dictionary
   - [Kafka Schema](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.schema_registry.Schema)
   - [Kafka RegisteredSchema](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.schema_registry.RegisteredSchema)
   - Kafka SchemaReference - missing from docs?

There are alternatives such as using kwargs such that users could continue to use the String representation of the schema if they had no need for references (I believe kwargs is similar in spirit to Java method overloading).   It is also possible to leverage the dynamic typing of Python to allow the schema arg to be either a String or a Schema (though this seems like it would be confusing).